core/keyserver/keyserver.go

```diff
@@ -666,6 +673,18 @@ func (s *Server) BatchQueueUserUpdate(ctx context.Context, in *pb.BatchQueueUser
 	return &empty.Empty{}, nil
 }
 
+func (s *Server) randLog(ctx context.Context, directoryID string) (int64, error) {
+	// TODO(gbelvin): Cache these results.
+	writable := true
+	logIDs, err := s.logs.ListLogs(ctx, directoryID, writable)
```

Comment:
What happens here if there are no writable logs? My naive (but experienced) intuition is that ListLogs will return an empty list with no error, and then panics below trying to find a random number for `rand.Intn(0)`.

Reply:
ListLogs will return codes.NotFound when there are no writable logs:
core/integration/storagetest/mutation_logs_admin.go